### DISCUSSION TITLE
parallelforeach  using designer

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5327] [Q&A]
How access iterated item of parallelforeach in parallelforeach body using designer?  Can you provide a JSON workflow sample

**Comment by glime-ai** [comment]
To access the iterated item of a `ParallelForEach` activity in Elsa using the designer, you would typically use JavaScript expressions to reference the current item. In the context of a `ParallelForEach` activity, the current item is accessible via a special variable that you can use within the body of the loop.

Here's a simplified example of how you might structure the JSON for a workflow that includes a `ParallelForEach` activity:

```json
{
  "name": "Parallel ForEach Example",
  "activities": [
    {
      "id": "parallel-foreach-activity",
      "type": "Elsa.Activities.ControlFlow.ParallelForEach",
      "properties": {
        "Items": {
          "expression": "[1, 2, 3]",
          "syntax": "Json"
        }
      },
      "activities": [
        {
          "id": "write-line-activity",
          "type": "Elsa.Activities.Console.WriteLine",
          "properties": {
            "Text": {
              "expression": "Current item: ${item}",
              "syntax": "JavaScript"
            }
          }
        }
      ]
    }
  ],
  "connections": [
    // Define connections between activities if necessary
  ]
}
```

In this JSON example, the `ParallelForEach` activity is configured with an `Items` property that defines the collection to iterate over. Within the body of the `ParallelForEach`, there is a `WriteLine` activity that uses a JavaScript expression to access the current item using `${item}`. The `item` variable is automatically available within the scope of the `ParallelForEach` activity and represents the current item being processed.

Please note that the actual property names and structure may vary depending on the version of Elsa you are using, and the example above is a generic representation. You should refer to the Elsa documentation or the JSON schema provided by Elsa for the exact structure required.

If you need a more specific example tailored to your use case, please provide additional details about the items you are iterating over and the actions you want to perform within the loop body.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
